spec/resources/groups_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'spec_helper'
+
+describe TableauApi::Resources::Groups, vcr: { cassette_name: 'groups' } do
+  include_context 'tableau client'
+
+  test_user_id = 'e1b91057-9cd9-4009-b6c9-cd18f1dc3fb4'
+
+  describe '#create' do
+    # http://onlinehelp.tableau.com/v9.0/api/rest_api/en-us/help.htm#REST/rest_api_ref.htm#Create_Group
+    it 'fails with a bad site_role' do
+      expect { client.groups.create(name: 'test', default_site_role: 'foo') }.to raise_error('invalid default_site_role')
+    end
+
+    it 'can create a group in a site' do
```

Comment:
I don't know if it would be meaningfully different, but you don't test specifying a default site role.